Missing OptionChoice as one of the possible values in basic_autocomplete's typehint

### Summary

The basic_autocomplete util function has a typehint that only allows an iterable of strings or integers or floats (or a callable/coroutine that returns one of these) but the actual code allows for the OptionChoice class by getting the "name" attribute. Therefore the typehint should also allow an instance of this class to be used in the iterable passed into the "values" argument.

### Reproduction Steps

1. Pass an iterable of `OptionChoice` instances to the `basic_autocomplete` util function
2. Run any static type-checker over the code (E.g. mypy or Pycharm's inbuilt checker
3. The argument will show as an error: `Mypy: Argument 1 to "basic_autocomplete" has incompatible type [arg-type]`

### Minimal Reproducible Code

```python
from typing import Iterable

import discord

choices: Iterable[discord.OptionChoice] = [discord.OptionChoice(name="Choice 1", value="val1"), discord.OptionChoice(name="Choice 2", value="val2")]

discord.utils.basic_autocomplete(choices)
```


### Expected Results

Static type-checkers (like mypy or Pycharm's inbuilt checker) should not show any type error inspections for the `values` argument to `basic_autocoplete`

### Actual Results

Static type-checkers (like mypy or Pycharm's inbuilt checker) show type error inspections for the `values` argument to `basic_autocoplete`

### Intents

None

### System Information

- Python v3.11.3-final              
- py-cord v2.4.1-final              
- aiohttp v3.8.4                    
- system info: Windows 10 10.0.19045

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

--- a/discord/utils.py
+++ b/discord/utils.py
@@ -142,6 +142,7 @@ if TYPE_CHECKING:
 
     from .abc import Snowflake
     from .commands.context import AutocompleteContext
+    from .commands.options import OptionChoice
     from .invite import Invite
     from .permissions import Permissions
     from .template import Template
@@ -156,6 +157,7 @@ if TYPE_CHECKING:
 else:
     cached_property = _cached_property
     AutocompleteContext = Any
+    OptionChoice = Any
 
 
 T = TypeVar("T")
@@ -1298,7 +1300,7 @@ def generate_snowflake(dt: datetime.datetime | None = None) -> int:
     return int(dt.timestamp() * 1000 - DISCORD_EPOCH) << 22 | 0x3FFFFF
 
 
-V = Union[Iterable[str], Iterable[int], Iterable[float]]
+V = Union[Iterable[OptionChoice], Iterable[str], Iterable[int], Iterable[float]]
 AV = Awaitable[V]
 Values = Union[V, Callable[[AutocompleteContext], Union[V, AV]], AV]
 AutocompleteFunc = Callable[[AutocompleteContext], AV]

